### ISSUE TITLE
About the input data shape

### ISSUE THREAD

**Issue by haozhi1817** [Luffy03/VoCo#20]
Hello, thanks for your great work!
 I am not familiar with MONAI's API,  so there are som confusions about the input shape.
I suppose every data will crop 16 no_overlap patch such as vit, and random crop 2 patch.
According to the code of dataloader and the function **concat_img** , I think the img shape is [batch_size *  4 * 4, 1, h, w, d], and the crops shape is [batch_size * 2, 1, h, w, d], and the labels shape is [batch_size, 2, 4*4]. Is it right?
https://github.com/Luffy03/VoCo/blob/1ae3fc9ece19031a74c54cb12c1c8f01715d7f90/models/voco_head.py#L173

**Comment by Luffy03** [comment]
First, many thanks for your attention to our work! Your understanding of the input shapes is correct.

**Comment by haozhi1817** [comment]
Thanks for your reply. 
Accoding to this line:
https://github.com/Luffy03/VoCo/blob/1ae3fc9ece19031a74c54cb12c1c8f01715d7f90/models/voco_head.py#L176
sw = batch_size * 4 * 4 // batch_size = 4 * 4
However, On the one hand,  I noticed you regard **sw** as num_of_random_crop throughout the entire code, so **sw** should be 2 .
On the other hand, Accoding to this line
https://github.com/Luffy03/VoCo/blob/1ae3fc9ece19031a74c54cb12c1c8f01715d7f90/models/voco_head.py#L194
if sw is 16, what is the meaning of 16 in **bases_stu_all[i * 16:(i + 1) * 16]**?

**Comment by haozhi1817** [comment]
After carefully reading the code， I think you regard **x_stu** as random crop patch features, and regard **base_stu** as regular crop patch features, 
so we should change 
https://github.com/Luffy03/VoCo/blob/1ae3fc9ece19031a74c54cb12c1c8f01715d7f90/models/voco_head.py#L175
to 
```python 
total_size = crops.size()[0]
```
and change https://github.com/Luffy03/VoCo/blob/1ae3fc9ece19031a74c54cb12c1c8f01715d7f90/models/voco_head.py#L180
to
```python
inputs = torch.cat([crops, img], dim=0)
```

**Comment by haozhi1817** [comment]
I am so sorry, your code is right. I am confused the meaning of img and crops

**Comment by Luffy03** [comment]
Dear researchers, our work is now available at [**Large-Scale-Medical**](https://github.com/Luffy03/Large-Scale-Medical), if you are still interested in this topic. Thank you very much for your attention to our work, it does encourage me a lot!
